mkosi/config.py

```diff
@@ -1186,6 +1194,20 @@ def __call__(
         parser.exit()
 
 
+S = TypeVar("S", bound=Any)
+T = TypeVar("T", bound=Any)
+
+
+def transform_dict_keys(obj: T, key_transformer: Callable[[S], S], exceptions: Sequence[type]) -> T:
+    if isinstance(obj, list):
+        return cast(T, [transform_dict_keys(e, key_transformer, exceptions) for e in obj])
+    elif isinstance(obj, dict) and type(obj) not in exceptions:
```

Comment:
Why doesn't `type(obj) is dict` work here so we don't need the `exceptions` argument?

Reply:
That's sufficient at the moment. `exceptions` was more explicit, since those were cases were we explicitly don't want capitalisation, but I'm fine with excepting every dict for now.